test/e2e/windows/gmsa.go

```diff
@@ -68,23 +68,7 @@ var _ = SIGDescribe("[Feature:Windows] [Feature:WindowsGMSA] GMSA [Slow]", func(
 				}
 
 				ginkgo.By("creating a pod with correct GMSA annotations")
-				f.PodClient().Create(pod)
-
-				ginkgo.By("waiting for the pod and its containers to be running")
```

Comment:
why are we removing this functionality? i assume that it was added because it was needed for some reason

Reply:
CreateSync() is a utility function that creates the pod and also waitsfor it to be running. see down a few lines.

Reply:
got it. so ideally we should merge this change in since it makes the test better. 
In addition, PR 11671 removed the test from 1.14.

anything else needed for this PR?

Reply:
We should merge this change because the original test was lacking this functionality. As stated in the issue, it did not wait for the pod to be in Running state.

Reply:
@adelina-t : I agree this is more elegant/less verbose, and thus better, but it seems to me the existing test was also waiting for the pod to be in a Running state (https://github.com/kubernetes/kubernetes/blob/dcbe588f76127cd7a3360e70f2260cd7c5ebbc73/test/e2e/windows/gmsa.go#L73-L87)?

Don't get me wrong, all for this change, I'd just like to know if I'm missing something here :)

Reply:
Both versions (existing and the suggested version) does the waiting. I do like the compact/standard version @adelina-t suggested because it's more elegant :-) So LGTM from me.

Reply:
@ddebroy @wk8 Indeed both versions should be doing the same. However, I've read more thoroughly, I've figured that the existing condition is wrong, here: https://github.com/kubernetes/kubernetes/blob/dcbe588f76127cd7a3360e70f2260cd7c5ebbc73/test/e2e/windows/gmsa.go#L75-L76

It basically checks if both there was an error in retrieving the pod **and** the pod state, which can't both be true.
It should be:
`
if err != nil || f.PodClinet() .... `

Reply:
Er. I feel dumb. Sorry...

I guess the other `Eventually` protecting the `exec` hid that when I ran it. Thank you for catching this.

Reply:
Yeah, if you've been running on local machine, probably it took next to nothing for the pod to be running, it just passed to the exec and never got a chance to notice. On testing clusters, depending on which node the pod lands, it takes some time for it to go into Running.

Reply:
Wasn't local, but wasn't a very busy cluster either. Thanks again!